src/github.com/matrix-org/dendrite/clientapi/routing/register.go

```diff
@@ -115,6 +116,11 @@ type registerRequest struct {
 
 	InitialDisplayName *string `json:"initial_device_display_name"`
 
+	// Prevent this user from logging in
+	// TODO: Ideally this would be a bool, but synapse has treated it as a number,
+	// so here we are.
+	InhibitLogin int `json:"inhibit_login"`
```

Comment:
I wonder if we should do something like https://stackoverflow.com/questions/30856454/how-to-unmarshall-both-0-and-false-as-bool-from-json so that we can accept both ints and bools